pkg/clusterapi/name.go

```diff
@@ -93,6 +93,11 @@ func ControlPlaneMachineTemplateName(clusterSpec *cluster.Spec) string {
 	return DefaultObjectName(fmt.Sprintf("%s-control-plane", clusterSpec.Cluster.Name))
 }
 
+// EtcdMachineTemplateName sets the default object name on the etcd machine template.
+func EtcdMachineTemplateName(clusterSpec *cluster.Spec) string {
```

Comment:
can we take only the eks-a cluster as input? it makes the dependency more clear and it's more reusable since you don't need the whole spec